### PR TITLE
Adds clang as second compiler for test cases

### PR DESCRIPTION
**Pull request by tbarabosch**
Even though clang is the first example, we could add more compilers in the future (e.g. the official ARM compilers).

### CODE REVIEW

File: test/acceptance/test_cwe476.py [hunk 1]
```diff
@@ -8,29 +8,35 @@ def setUp(self):
...
         self.assertEqual(res, expect_res)
 
-    def test_cwe476_01_x64(self):
+    def test_cwe476_01_x64_gcc(self):
         expect_res = 1
-        res = cwe_checker_testlib.execute_and_check_occurence(self.target, self.target, 'x64', self.string)
+        res = cwe_checker_testlib.execute_and_check_occurence(self.target, self.target, 'x64', 'gcc', self.string)
+        self.assertEqual(res, expect_res)
+
+    @unittest.skip("FIXME")
```
Comment (Enkelmann) [inline test/acceptance/test_cwe476.py]: I should take a look at this.

Comment (Enkelmann) [reply test/acceptance/test_cwe476.py]: clang seems to optimze away the relevant calls for the check. Is there a "do not optimize anything"-flag for clang that we could use?

Comment (tbarabosch) [reply test/acceptance/test_cwe476.py]: we could build the binaries with -o0

**Comment by tbarabosch** [comment]
I am going to look into the x86_gcc issue. But I suppose it's a problem with dockcross....

File: test/acceptance/test_cwe476.py (hunk 1 above)
Comment (Enkelmann) [reply test/acceptance/test_cwe476.py]: If it is not enabled by default, we should do that.

**Comment by tbarabosch** [comment]
I did a clean build of all artificial examples on this branch. I was able to build for all arches including x86. Could you please check your dockcross installation?

**Comment by Enkelmann** [comment]
OK, then it is probably a problem on my machine. Merging.

**Pull request merged**